plugins/inputs/execd/execd_posix.go

```diff
@@ -6,30 +6,33 @@ import (
 	"fmt"
 	"io"
 	"os"
-	"os/exec"
 	"syscall"
 	"time"
 
 	"github.com/influxdata/telegraf"
 )
 
 func (e *Execd) Gather(acc telegraf.Accumulator) error {
-	if e.cmd == nil || e.cmd.Process == nil {
+	if e.process == nil || e.process.Cmd == nil {
 		return nil
 	}
 
+	osProcess := e.process.Cmd.Process
+	if osProcess == nil {
+		return nil
+	}
 	switch e.Signal {
 	case "SIGHUP":
-		e.cmd.Process.Signal(syscall.SIGHUP)
+		osProcess.Signal(syscall.SIGHUP)
 	case "SIGUSR1":
-		e.cmd.Process.Signal(syscall.SIGUSR1)
+		osProcess.Signal(syscall.SIGUSR1)
 	case "SIGUSR2":
-		e.cmd.Process.Signal(syscall.SIGUSR2)
+		osProcess.Signal(syscall.SIGUSR2)
 	case "STDIN":
-		if osStdin, ok := e.stdin.(*os.File); ok {
+		if osStdin, ok := e.process.Stdin.(*os.File); ok {
```

Comment:
Is this right, it doesn't seem like it would ever be an os.File.

Reply:
yeah. I actually inspected the underlying type to figure out what it was. I was surprised, but it's definitely an *os.File.